### ISSUE TITLE
composer cache repo directory name

### ISSUE THREAD

**Issue by IlmLV** [composer/composer#7439]
My `composer.json`:

```json
...
    "repositories": [
        {
            "type": "vcs",
            "url": "https://gitlab+deploy-token-4:..token..@..domain...lv/composer/http-client.git"
        }
    ],
...
```
everything works fine, but I'm worried about token stored in local appdata folder as repo directory name, token should be removed from folder name

![image](https://user-images.githubusercontent.com/1309998/42208840-e26f7c40-7eb5-11e8-957e-201272319022.png)

**Comment by alcohol** [comment]
@Seldaek do you think we could get away with simply stripping credentials from the url before attempting to use it as a directory name for caching purposes? I am trying to think of a scenario where that could result in naming clashes/conflicts, but cannot really think of one.

**Comment by Seldaek** [comment]
Fixed for 2.0 by https://github.com/composer/composer/pull/9155
